src/model/keychain.js

```diff
@@ -136,3 +148,94 @@ export async function hasKey(key) {
   }
   return null;
 }
+
+export async function restoreBackupIntoKeychain(backedUpData) {
+  // Access control config per each type of key
+  const publicAccessControlOptions = {
+    accessible: ACCESSIBLE.ALWAYS_THIS_DEVICE_ONLY,
+  };
+
+  const privateAccessControlOptions = await getPrivateAccessControlOptions();
+
+  await Promise.all(
+    Object.keys(backedUpData).map(async key => {
+      const value = backedUpData[key];
+      let accessControl = publicAccessControlOptions;
+      if (endsWith(key, seedPhraseKey) || endsWith(key, privateKeyKey)) {
+        accessControl = privateAccessControlOptions;
+      }
+      if (typeof value === 'string') {
+        return saveString(key, value, accessControl);
+      } else {
+        return saveObject(key, value, accessControl);
+      }
+    })
+  );
+
+  // Save the migration flag
+  // to prevent this flow in the future
+  await saveString(
+    oldSeedPhraseMigratedKey,
+    'true',
+    publicAccessControlOptions
+  );
+
+  return true;
+}
+
+// Attempts to save the password to decrypt the backup from the iCloud keychain
+export async function saveBackupPassword(password) {
+  try {
+    await setSharedWebCredentials('rainbow.me', 'Backup Password', password);
+  } catch (e) {
+    logger.sentry('Error while backing up password');
+    captureException(e);
+  }
+}
+
+// Attempts to fetch the password to decrypt the backup from the iCloud keychain
+export async function fetchBackupPassword() {
+  try {
+    const results = await requestSharedWebCredentials();
+    return results?.password || null;
+  } catch (e) {
+    logger.sentry('Error while fetching backup password', e);
+    captureException(e);
+    return null;
+  }
+}
+
+export async function wipeKeychain() {
+  try {
+    const results = await loadAllKeys();
+    await Promise.all(
+      results.map(result => resetInternetCredentials(result.username))
+    );
+    logger.log('keychain wiped!');
+  } catch (e) {
+    logger.sentry('error while wiping keychain');
+    captureException(e);
+  }
+}
+
+export async function getPrivateAccessControlOptions() {
+  let res = {};
+  // This method is iOS Only!!!
+  const canAuthenticate = await canImplyAuthentication({
```

Comment:
I know this is unlikely to be an issue unless on Android, but can we just surround it with a try/catch (letting the returned private access control options still be empty)